src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -40,6 +43,13 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 return;
             }
 
+            // if there is a const keyword, the refactoring shouldn't show as interpolated string is not const string
+            var declaration = top.FirstAncestorOrSelf<SyntaxNode>(syntaxFacts.IsMethodLevelMember);
+            if (declaration != null && IsConstMemberDeclaration(declaration, syntaxFacts))
+            {
+                return;
+            }
```

Comment:
can actually make this simpler as we can do the following:

```c#
var generator = SyntaxGenerator.GetGenerator(document);
if (generator.GetModifiers(declaration).IsConst) ...
```

Then you can remove the abstract method and subclass overrides.

Reply:
Why is this on SyntaxGenerator?

Reply:
in general, because it's teh public surface area for simplified querying/editing/generator of syntax.